packages/runtimes/js/src/helpers/bundle-url.js

```diff
@@ -32,7 +32,11 @@ function getBaseURL(url) {
 
 // TODO: Replace uses with `new URL(url).origin` when ie11 is no longer supported.
 function getOrigin(url) {
-  let matches = ('' + url).match(/(https?|file|ftp):\/\/[^/]+/);
+  url = '' + url;
```

Comment:
Instead of mutating the `url` arg, maybe try `if (typeof url === 'string' && url[0] === '/')` and keep the `('' + url).match(...` below.

Reply:
My assumption was that `(''+url)` is an implicit conversion to string? I'm unfamiliar with this particular idiom - I'd much rather `if (typeof url !== 'string') { throw new Error('you accidentally a url') }` and not append it to an empty string at all (memory copy optimization?)

Reply:
I believe that is what it is doing. `''+undefined === "undefined"```, so if `url` is undefined or null, etc., you are not going to get a match anyway. But, I suppose it ensures that `.match` never fails.